fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListViewController.swift

```diff
@@ -31,46 +55,226 @@ final class CustomValidatorListViewController: UIViewController, ViewHolder {
+
+            updateProceedButton()
         }
     }
 }
 
+// MARK: - CustomValidatorListViewProtocol
+
 extension CustomValidatorListViewController: CustomValidatorListViewProtocol {
-    func reload(with viewModel: [CustomValidatorCellViewModel]) {
-        cellViewModels = viewModel
-        rootView.tableView.reloadData()
+    func reload(_ viewModel: CustomValidatorListViewModel, at indexes: [Int]? = nil) {
+        cellViewModels = viewModel.cellViewModels
+        headerViewModel = viewModel.headerViewModel
+
+        if let indexes = indexes {
+            let indexPaths = indexes.map {
+                IndexPath(row: $0, section: 0)
+            }
+            rootView.tableView.reloadRows(at: indexPaths, with: .none)
+        } else {
+            rootView.tableView.reloadData()
+        }
+
+        selectedValidatorsCount = cellViewModels.filter {
+            $0.isSelected
+        }.count
+
+        updateFillRestButton()
+        updateDeselectButton()
+        updateProceedButton()
+    }
+
+    func setFilterAppliedState(to applied: Bool) {
+        filterIsApplied = applied
+        updateClearFiltersButton()
+    }
+
+    func updateHeaderViewModel(to viewModel: TitleWithSubtitleViewModel) {
+        headerViewModel = viewModel
     }
 }
 
+// MARK: - UITableViewDataSource
+
 extension CustomValidatorListViewController: UITableViewDataSource {
     func tableView(_: UITableView, numberOfRowsInSection _: Int) -> Int {
         cellViewModels.count
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCellWithType(CustomValidatorCell.self)!
+        cell.delegate = self
+
         let viewModel = cellViewModels[indexPath.row]
         cell.bind(viewModel: viewModel)
-        cell.infoButton.addTarget(self, action: #selector(handleValidatorInfo), for: .touchUpInside)
+
         return cell
     }
 }
+
+// MARK: - UITableViewDelegate
+
+extension CustomValidatorListViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        presenter.changeValidatorSelection(at: indexPath.row)
+    }
+
+    func tableView(_: UITableView, heightForHeaderInSection _: Int) -> CGFloat {
+        guard headerViewModel != nil else { return 0 }
+        return 26.0
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection _: Int) -> UIView? {
+        guard let headerViewModel = headerViewModel else { return nil }
+        let headerView: CustomValidatorListHeaderView = tableView.dequeueReusableHeaderFooterView()
+        headerView.bind(viewModel: headerViewModel)
+        return headerView
+    }
+
+    func tableView(_: UITableView, heightForFooterInSection _: Int) -> CGFloat {
+        100.0
```

Comment:
Why do we need this dynamic footer? If we need an inset then we could use contentInset property

Reply:
You are right. Fixed that on recommended validators screen as well